### PR TITLE
fix(typescript_indexer): do not give anonymous names to binding patterns

### PR DESCRIPTION
**Pull request by ayazhafiz**
Variable declarations in object and array binding patterns like `const [a, b] = [0, 0]` are currently given a signature with an superfluous anonymous block scope, so `a` has a signature like `anon0.a`. Fix this so `a` has a signature of just `a`.

_edit_: formatting

### CODE REVIEW

**Comment by ayazhafiz** [comment]
cc @evmar, @kythe

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -442,7 +442,15 @@ class Visitor {
             }
             parts.push(part);
           } else {
-            // TODO: handle other declarations, e.g. binding patterns.
```
[the remark this answers is not in the document]
Comment (ayazhafiz) [reply kythe/typescript/indexer.ts]: Done.

**Comment by ayazhafiz** [comment]
@kythe could you review?

**Comment by salguarnieri** [comment]
Please update the branch and then I'll merge

**Pull request merged**